### ISSUE TITLE
[a11y] 11.1.4.3 Contrast (minimum)

### ISSUE THREAD

**Issue by tschiebel** [owncloud/android#4364]
audit: https://infinite.owncloud.com/s/QtwuiGHikwFjozV

https://www.etsi.org/deliver/etsi_en/301500_301599/301549/03.02.01_60/en_301549v030201p.pdf
https://github.com/BIK-BITV/BIK-App-Test/tree/main/Pr%C3%BCfschritte/de

Note 1: For texts (also in the focused state) under 18 points (or bold text below 14 points), the contrast ratio between foreground and background color should be at least 4.5:1. For large fonts, the contrast ratio should be 3:1 or greater.
Note 2: Since the contrast measurements in the test were carried out using screenshots the test results may differ from the actual color values. It is possible that the colors are displayed differently on the screenshot than on the screen of the device. For this reason, the test results should be regarded as guide values.

### Personal Space

- The labelling of the "Browse folder" input field does not meet the contrast requirements for texts, as the contrast ratio is only approx. 2.9:1. [MOVED TO owncloud/android/issues/4365]
![image](https://github.com/owncloud/android/assets/132444655/6904508f-a7bf-4296-ac4e-ae30df436937)

- [x] Both the size information and the time information on the page do not fulfill the contrast requirements for texts, as the contrast ratio is currently well below 4.5:1. The text at the bottom of the page, which shows the number of directories and files also does not meet the contrast requirements for text.
![image](https://github.com/owncloud/android/assets/132444655/125b78ac-e373-4e6b-aa1c-805ce9a4efb7)

- [x] The "Manage accounts" button has a visual label in the form of a letter that does not meet the contrast requirements and is barely recognizable.
![image](https://github.com/owncloud/android/assets/132444655/5cc5d7a3-6773-447e-99d7-a6939d1d32cb)

### Menu

- [x] Although the text with the storage specifications fulfills the contrast requirements, the contrast ratio may be just below the recommended value. To ensure that all users can read the text clearly and distinctly, it is advisable to improve the contrast even further.
![image](https://github.com/owncloud/android/assets/132444655/2259972b-fee9-49c1-83ca-15956b138632)

### Share

- [x] The size specifications and the note under "Public links" both do not meet the contrast requirements for text.
![image](https://github.com/owncloud/android/assets/132444655/2298d58e-bc32-4ec1-9067-4f13ab2d54c8)

### Edit shared Link

- [x] The date does not meet the contrast requirements for text.
![image](https://github.com/owncloud/android/assets/132444655/42775057-f557-4547-8f51-0d582c99e524)

### Spaces

- The label of the "Search folder" input field does not meet the contrast requirements for text, as the contrast ratio is only approx. 2.9:1. [MOVED TO owncloud/android/issues/4365]
![image](https://github.com/owncloud/android/assets/132444655/302369f5-87ff-44fa-b9aa-c59a16a615ec)

**Comment by jesmrec** [comment]
- Learn about contrast measure (what does 3:1 mean?)
- Check if back and fore colors are part of the branding.
- Make colors brandable or for everyone in the vanilla app. ¿?

**Comment by Aitorbp** [comment]
Regarding the colors that need to be changed to meet the contrast requirements, the following research has been carried out:
In the first instance, we wanted to use the colors that Android Studio has in the system (`/Library/Android/sdk/platforms/android-33/data/res/values/colors.xml`) and they met the contrast requirements.
But these colors have two problems:

1. Some are private: When you try to access them, Android Studio does not let you compile the application because they are private and cannot be used by us.
2. It requires an API higher than 31: This restricts users with devices lower than Android 12 from having these colors. If an API lower than 31 is used, Android will set a default color. In this case, using the color `system_accent1_500`, Android set this purple color on the texts:

![Image](https://github.com/owncloud/android/assets/34248951/6dcc4e5d-f0e9-4924-a471-319680a1fd55)


In conclusion, I would not use the colors found in the system due to these restrictions and instead, I would take the color that meets the contrast requirements and put it in our `colors.xml` file, so that we would avoid these restrictions.

If we use the `High Contrast accessibility action` from the device settings with this color, the behavior would be the same as if we took the color from the system.

Without high contrast fonts:

![Image](https://github.com/owncloud/android/assets/34248951/8db8db80-eb87-48c4-80dc-72bb478e68ea)

With high contrast fonts:

![Image](https://github.com/owncloud/android/assets/34248951/49aae6f0-b35d-4f48-8c74-ce4a533178b7)

**Comment by Aitorbp** [comment]
This issue will be blocked until we know how to proceed with the toolbar elements. The background of the toolbar is brandable (`actionbar_start_color`) and with the current color of the image of the issue it does not meet the accessibility requirements. If these values ​​were left as default, the toolbar would meet these requirements.
A possible solution is to set the color of the hint text and the cross of the toolbar as a brandable parameter, so that clients can adjust these values ​​to meet the contrast requirements. The problem is that in this case we would be leaving the contrast in the hands of their branding.

**Comment by TheOneRing** [comment]
Your approach sounds sensible, please continue.
Could you please open an issue with ownBrander to amend the description for the brandable colors?

**Comment by jesmrec** [comment]
> Could you please open an issue with ownBrander to amend the description for the brandable colors?

sure, i will take care

**Comment by JuancaG05** [comment]
> * [ ]  The "Manage accounts" button has a visual label in the form of a letter that does not meet the contrast requirements and is barely recognizable.
>   ![image](https://private-user-images.githubusercontent.com/132444655/319988692-5cc5d7a3-6773-447e-99d7-a6939d1d32cb.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5MjA2NjIsIm5iZiI6MTcxOTkyMDM2MiwicGF0aCI6Ii8xMzI0NDQ2NTUvMzE5OTg4NjkyLTVjYzVkN2EzLTY3NzMtNDQ3ZS05OWQ3LWE2OTM5ZDFkMzJjYi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwMlQxMTM5MjJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZWRjMmMwMzVlNWMyOWZjNTJlODI2OGNhMjZkM2QzOGQ2OWIyMWZlYWE4OGJhZGE4YmYyNjkxZjNmZGQxNWEzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.2lIYzMslrb3Mvyzmtrdf9iZoEJgSv0OjAlv98e2WIHc)

This one won't be addressed since the avatar is not generated in the app, it comes from the server

**Comment by JuancaG05** [comment]
> * The labelling of the "Browse folder" input field does not meet the contrast requirements for texts, as the contrast ratio is only approx. 2.9:1.
>   ![image](https://private-user-images.githubusercontent.com/132444655/319988456-6904508f-a7bf-4296-ac4e-ae30df436937.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk5MjExMDMsIm5iZiI6MTcxOTkyMDgwMywicGF0aCI6Ii8xMzI0NDQ2NTUvMzE5OTg4NDU2LTY5MDQ1MDhmLWE3YmYtNDI5Ni1hYzRlLWFlMzBkZjQzNjkzNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzAyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwMlQxMTQ2NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1kY2Q0YWE4MzgyMjY5YzdlYmYzYmMzMWI4Y2JjZmU0MTRkN2Y3MmNkMTliMjk0MjdhNjllNTgzNWZmMjQwZDcyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.d0INc6JAi-_tzYS8-ne2joB7-wuR7CIKcJpg77tYIVg)

This one will be addressed in https://github.com/owncloud/android/issues/4365, where more changes to the toolbar will be applied

**Comment by jesmrec** [comment]
About ownBrander stuff. There are not many colors to brand there. Not all of them could be affected by non-matching-contrast colors. On the bottom, i am listing the ones that are affected which description should be updated in oB, if all of you are right.

### Splash screen background

Splash screen is built with that color on the background and the app icon in the foreground. Here, contrast is not affected since accessibility is not required (no information for dictation or reading)


### Login screen

Almost everything is brandable in the login view. In that case, it's in the customer's hand to choose colors with correct contrast. On the other hand, there is no accessibility report regarding login view. 

- Background color
- Text id and password
- Hint on id and password fields
- Background on the login button
- Text on the login button
- Connection status
- Error message

**Accessibility contrast**: delegated to customer.

### Primary color

Well, after checking a little deeper, i'm disappointed with such option. Apart of the FAB, i did not find any place in the app where the "primary color" is set. 

**Accessibility contrast**: The icons inside the buttons are white, so, the color should not be close to the white.

![Screenshot 2024-07-02 at 14 03 43](https://github.com/owncloud/android/assets/14894746/8c133e24-03b3-4093-b11f-93543cafdbf5)

### Primary text color

Text color used within the same buttons as the primary color. I only found this as tint color for the bottom bar icons

**Accessibility contrast**: Should match with the `Top bar color` in the background

![Screenshot 2024-07-02 at 14 03 31](https://github.com/owncloud/android/assets/14894746/4f97de9d-2fe7-46fb-9a06-9bf2a1edc8d2)

### Top bar color

Color of the top bar. But used everywhere in the app: in the search bar, order bar, bottom bar. Also, as color in the details view. That seems to be the real primary color of the app.

**Accessibility contrast**: The search bar, as we already know, will need a tweak. This color is also used in the details view. I don't like this but we can live with it if the color it not close to the white background.

|![Screenshot 2024-07-02 at 14 03 19](https://github.com/owncloud/android/assets/14894746/ffb633e2-323f-4535-be73-114a09515174)|![Screenshot 2024-07-02 at 14 04 01](https://github.com/owncloud/android/assets/14894746/97f57943-88fb-4d92-81b8-666ea5aee912)|

### General menu header 

Background color on the drawer. 

**Accessibility contrast**: Text is white, so, the color should not be close to the white.

![Screenshot 2024-07-02 at 14 07 15](https://github.com/owncloud/android/assets/14894746/e8fa53b9-1b89-4efc-9f74-95890eef9b70)

### Color of the space card (oCIS) 

Color of the space card.

**Accessibility contrast**: view background is white and text inside is black. Something to be careful of, but in the hands of the customers.

![Screenshot 2024-07-02 at 14 09 28](https://github.com/owncloud/android/assets/14894746/acceba19-4a79-4ba5-b8c6-0b8aba8a6fea)

### Where should we warn customers about contrast requirements:

- Login Screen
- Primary text color (because collisions with top bar)
- Top bar, because some text is tinted with it

The others, that can collide with white and black could be also involved in problems, but it sounds more weird to have these cases... open to ideas here.
